plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportStateChangedObserver
+import com.mapbox.maps.plugin.viewport.experimental.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransitionOptions
+import java.util.concurrent.CopyOnWriteArrayList
+
+/**
+ * Mapbox default implementation for [ViewportPlugin].
+ */
+class ViewportPluginImpl : ViewportPlugin {
+  private val registeredViewportStates = CopyOnWriteArrayList<ViewportState>()
+  private var currentState: ViewportState = ViewportState.IDLE
+
+  override val state: ViewportState
+    get() = currentState
+
+  override fun registerViewportState(state: ViewportState) {
+    registeredViewportStates.add(state)
+  }
+
+  override fun unregisterViewportState(state: ViewportState) {
+    registeredViewportStates.remove(state)
+  }
+
+  override fun requestCameraToState(
```

Comment:
The name feels a little awkward to me. What about `transitionToState`

Reply:
This follows current NavigationCamera naming, we could rename it to transitionToState for now.